lib/turn_junebug_expressway/agent.ex

```diff
@@ -0,0 +1,45 @@
+defmodule TurnJunebugExpressway.MessageRecipientIdCache do
+  use GenServer
+
+  def start_link(options \\ []) do
+    {name, options} = Keyword.pop(options, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, options, name: name)
+  end
+
+  def put(
+        pid,
+        key,
+        value,
+        ttl \\ get_env(:agent, :ttl) |> String.to_integer()
+      ) do
+    GenServer.call(pid, {:put, key, value, ttl})
+  end
+
+  def get(pid, key) do
+    GenServer.call(pid, {:get, key})
+  end
+
+  # GenServer callbacks
+
+  def init(_) do
+    state = %{}
+    {:ok, state}
+  end
+
+  def handle_call({:put, key, value, ttl}, _from, state) do
+    Process.send_after(self(), {:expire, key}, ttl)
```

Comment:
Doing it this way means that if we set the same key multiple times we're not extending the lifetime -- each call will schedule a new expiry. I don't think that's a problem for message ids, but we may want to add a comment noting that behaviour.